src/models/message_status.ts

```diff
@@ -64,12 +64,26 @@ export const getMessageStatusUpdater = (
   messageId: NonEmptyString
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 ): MessageStatusUpdater => status =>
-  messageStatusModel.upsert({
-    kind: "INewMessageStatus",
-    messageId,
-    status,
-    updatedAt: new Date()
-  });
+  pipe(
+    messageStatusModel.findLastVersionByModelId([messageId]),
```

Comment:
Why do we have to retrieve last version before upsert?

Reply:
Because we cannot upsert "isRead" and "isArchived" without knowing their previous values

Reply:
AFAIK MessageStatusUpdater is called within the ProcessMessage Workflow in order to track if a message is correctly processed, at the same time a citizen cannot trigger a change status (isPending -> true). Btw LGTM to let the updater agnostic related to workflow usage.

Reply:
Yes, I saw it, but I prefer to be prepared on new statuses and avoid future mistakes